pages/api/todos/[id].js

```diff
@@ -0,0 +1,32 @@
+import { deleteToDo, getToDoById, updateToDo } from "../../../services/toDoService";
+
+export default async function handler(req, res) {
+  const {
+    query: { id, destinationId },
+    method,
+  } = req;
+
+  switch (method) {
+    case "GET":
+      const toDo = await getToDoById(id);
+      if (toDo.error) return res.status(toDo.status).json({ error: toDo.error });
+      res.status(200).json(toDo);
+      break;
+
+    case "PATCH":
+      const newToDos = await updateToDo(id, req.body);
+      if (newToDos.error) return res.status(newToDos.status).json({ error: newToDos.error });
+      res.status(201).json(newToDos);
+      break;
+
+    case "DELETE":
+      const deletedToDo = await deleteToDo(id, destinationId);
```

Comment:
Why do you need the destinationId here? The destinationId is simply a field in the ToDo Document. YOu should be able to delete a todo item, regardless of a value of any of its fields.